src/lib.rs

```diff
@@ -343,12 +343,18 @@ mod tests {
         assert_eq!(entry.object, Some("allocatorsysmem0".to_string()));
     }
 
-    #[test]
-    fn color() {
-        let f = File::open("test-logs/color.log").expect("Failed to open log file");
+    fn parse_file(f: File) -> (Entry, usize) {
         let mut parsed = parse(f);
 
         let entry = parsed.next().expect("First entry missing");
+        // FIXME: the +1 shouldn't be needed here.
```

Comment:
Why not? You return the first element of the iter and count the rest right?

Reply:
Oh yeah, facepalm :)